client/app/components/compose_toolbox.coffee

```diff
@@ -5,31 +5,49 @@
 module.exports = ComposeToolbox = React.createClass
     displayName: 'ComposeToolbox'
 
+    getInitialState: ->
+        action: null
 
-    render: ->
-        if @props.sending
-            labelSend = t 'compose action sending'
-        else
-            labelSend = t 'compose action send'
+    getLabel: ->
+        className = if @state.action is 'send' then 'sending' else 'send'
+        t 'compose action ' + className
+
+    save: ->
+        @triggerAction 'save'
+
+    send: ->
+        @triggerAction 'send'
+
+    delete: ->
+        @triggerAction 'delete'
 
+    cancel: ->
+        @triggerAction 'cancel'
+
+    triggerAction: (type) ->
+        @setState action : type
+        if (callback = @props[type]) and _.isFunction callback
```

Comment:
Better to use `listener` than `callback` here. You can also use `@props[type]?()`. 
I feel `triggerAction` might be too much factorisation, as I often search for `props.xxxxx` in a component to see how its used, may be it would be cleaner to just copy paste 
```
@setState action: 'xxxx'
@props.xxxxx?()
``` 
in the functions above ?